src/libstore/build.cc

```diff
@@ -2380,12 +2380,44 @@ void DerivationGoal::runChild()
 
             sandboxProfile += "(allow file-read* file-write-data (literal \"/dev/null\"))\n";
 
+            sandboxProfile += "(allow ipc-posix-shm*)\n";
+
+            sandboxProfile += "(allow mach-lookup\n"
+                "\t(global-name \"com.apple.SecurityServer\")\n"
+                ")\n";
+
+            sandboxProfile += "(allow file-read*\n"
+                "\t(literal \"/dev/dtracehelper\")\n"
+                "\t(literal \"/dev/tty\")\n"
+                "\t(literal \"/System/Library/CoreServices/SystemVersion.plist\")\n"
```

Comment:
Yeah, probably would be easier to manage there than here (given repo permissions). The `ipc-posix-shm*` stuff and friends seems fine to leave here. Or we can just split this stuff out into a file that can be managed more flexibly than modifying a source file every time :smile: